### ISSUE TITLE
macOS version still at 1.5.3

### ISSUE THREAD

**Issue by qwertychouskie** [Neverball/neverball#151]
1.6 has been out since 2014...

**Comment by ChallengeSY** [comment]
Regrettably, we do not currently have an individual willing and **able** to build version 1.6.0.

**Comment by parasti** [comment]
I assume Nuncabola is more than enough for any OSX folk.

**Comment by ChallengeSY** [comment]
Yes... except for the lack of a "Nuncaputt".

If I ever bring back Random Courses (not likely while the 16 course limit is in place), I would like to be able to offer just one package instead of **two** for each course "tournament".

**Comment by camthesaxman** [comment]
If I remember correctly, the reason why Neverball 1.6.0 wasn't released for Mac OS X was because of a bug in SDL2 that prevented full screen from working properly. Has anyone tried to build it recently for Mac OS X recently and see if it works properly?

**Comment by parasti** [comment]
> Neverball 1.6.0 wasn't released for Mac OS X was because of a bug in SDL2 that prevented full screen from working properly.
That is incorrect. It wasn't released for Mac because there was nothing to release. Meaning, nobody ever built 1.6.0 for Mac and gave us a build to release.

**Comment by qwertychouskie** [comment]
Travis provides builds for MacOS: https://docs.travis-ci.com/user/reference/osx/

Maybe this can be used?

**Comment by xif-fr** [comment]
Hi, I successfully built Neverball 1.6.0 on OSX 10.9. There was nothing particular to do, appart messing around in Xcode and installing SDL2. But my build is very messy, so I can't release it like that. If I have time, I'll make a proper build and update the Xcode projet.

Thanks for this great game.

**Comment by xif-fr** [comment]
Ok, I'll do it. Could take days or months. Do I have to keep the 1.6.0 port as close as possible to the old one, or can I do big changes as long as the distributed build is clean and tidy ? I may have to drop support up to OSX 10.6.8.

**Comment by parasti** [comment]
You don't have to preserve anything of the old port.

**Comment by camthesaxman** [comment]
Some day I will buy a Mac, maybe...

**Comment by qwertychouskie** [comment]
I actually have a 2010ish iMac now.  It randomly shuts off, but is sufficient for building (I used it to do some work on the SuperTuxKart macOS build).  However, I would much rather package a new 1.6.1 than an old 1.6.0.  @parasti @rlk Are there any plans to have a 1.6.1 bugfix release soon?  Maybe after https://github.com/Neverball/neverball/pull/157 and https://github.com/Neverball/neverball/pull/189 (and https://github.com/Neverball/neverball/pull/188?  Any reason why this isn't merged?) are merged we could work on this.  I'm happy to help where possible.

**Comment by xif-fr** [comment]
If have enough motivation to make the OSX build, it would be great. Otherwise, I plan to do it this month. It seams to me that the Xcode project and packaging needs quite a bit of rework/cleanup.

**Comment by xif-fr** [comment]
It's nearly done. I have still the packaging to do. I could modify the existing scripts for creating the installer, which is a pain, or use the nice open-source software [Packages](http://s.sudre.free.fr/Software/Packages/about.html).

Moreover, I have a proposition for the distribution on OSX :
- As before, an installer which install the `data` directory in `/Library/Application Support/Neverball`, in common for Neverball and Neverputt
- A standalone Neverball, with the `data` directory embedded in Neverball.app, minus the Neverputt maps
- A standalone Neverputt, with the `data` directory embedded in Neverball.app, minus the Neverball maps

The standalone versions are wasting some space because common files are no longer in common, but it has the advantage of being much more easily distributed (no installer needed, and could be distributed on the app store).

What do you think about this ?

**Comment by qwertychouskie** [comment]
Personally I think portable installs is better than having to use an installer.

**Comment by xif-fr** [comment]
Here is a standalone version, statically linked with libpng/jpeg/vorbis/ogg, data included, and backward-compatible up to OS X 10.6.8 (the last version supported by SDL2) : [https://www.xif.fr/uploads/Neverball.app.zip](https://www.xif.fr/uploads/Neverball.app.zip). It has been tested on a 10.6.8 and on a 10.9.5, both heavily modified, so I would need feedback on vanilla systems, and more recent macOS versions. I'm currently cleaning up the build files and re-writing the build procedure.

**Comment by xif-fr** [comment]
And here is Neverputt, built the same way : [https://www.xif.fr/uploads/Neverputt.app.zip](https://www.xif.fr/uploads/Neverputt.app.zip)

**Comment by qwertychouskie** [comment]
@xif-fr Seems to work on my 10.13.6 iMac with a somewhat clean OS (only a few things installed via Brew for building SuperTuxKart).

**Comment by qwertychouskie** [comment]
Any reason these build can't be found at https://neverball.org/download.php?  Would be nice to finally close this issue. :)

**Comment by xif-fr** [comment]
Well, these builds were a bit hacked together. But I think they are clean enough to be released, at least with a link to the older release in case this build does not work.
Indeed, in the meantime, I almost completely switched to Linux after being fed up with my OSX system. Maybe, one day, I will finish what I began…

**Comment by qwertychouskie** [comment]
The build still isn't up on https://neverball.org/download.php, would be nice to see it there soon :)

**Comment by limdingwen** [comment]
> Here is a standalone version, statically linked with libpng/jpeg/vorbis/ogg, data included, and backward-compatible up to OS X 10.6.8 (the last version supported by SDL2) : https://www.xif.fr/uploads/Neverball.app.zip. It has been tested on a 10.6.8 and on a 10.9.5, both heavily modified, so I would need feedback on vanilla systems, and more recent macOS versions. I'm currently cleaning up the build files and re-writing the build procedure.
> 
> Neverputt is coming too.
> 
> Is there any legal issue with static linking of libraries ?

Thank you for your builds, and thank you for keeping them online for so long. I can confirm it works on my M1 Mac, though you need to right-click open.

Here's a mirror:

[NeverBall build by xif.fr](https://ignisincendio.me/mirror/games/Neverball.app.zip)
[NeverPutt build by xif.fr](https://ignisincendio.me/mirror/games/Neverputt.app.zip)

**Comment by limdingwen** [comment]
> I am going to try to get a [Homebrew](https://brew.sh) formulae written to make building/installing Neverball via [Homebrew](https://brew.sh) is easier. It would be neat to get this building natively on the new Arm Macs(the x86 posted here version works fine on M1 Macs though).

I'm a bit unsure on how Homebrew works, but given that .apps bundles don't have any external dependencies, I'm unsure of how useful a package manager is. Yes, I know I'm calling the whole cask thing into question :P 

It doesn't seem like they perform any sort of hosting for builds?
